### ISSUE TITLE
App's request for commons.js returns root HTML page when worker instantiated

### ISSUE THREAD

**Issue by msakrejda** [universse/gatsby-plugin-workerize-loader#9]
Hello,

Thanks for writing this plugin! I'm running into a weird issue that I've narrowed down to instantiating a worker when using this plugin. I have a Gatsby app that loads several JS files including commons.js via its root HTML. I've configured `gatsby-plugin-workerize-loader` according to the instructions, and the app builds fine, but if I instantiate the worker, the fetch request for commons.js returns the root HTML instead of returning the correct contents (other JavaScript files requested concurrently seem to load fine). If I comment out the line instantiating the worker, the app works fine again (without the worker, obviously).

I'm using Gatsby 2.29.3 and gatsby-plugin-workerize-loader 1.5.0 . Any ideas what could be happening?

**Comment by msakrejda** [comment]
I replaced my (emscripten-generated) actual worker code with a dummy implementation that just errors out, and that seems to work fine, so pretty sure there's not actually a `gatsby-plugin-workerize-loader` bug here, though if you have any ideas, I'd love to hear them. Otherwise feel free to close this and sorry for the noise.

**Comment by phuoccss** [comment]
Do you have a minimal reproduction?

**Comment by msakrejda** [comment]
I don't, but I did figure out my issue: I had to set [USE_ES6_IMPORT_META=0](https://github.com/emscripten-core/emscripten/blob/main/src/settings.js#L1172-L1176) in the Emscripten compilation. It looks like the webpack config Gatsby 2 is using does not support `import.meta`, and I was getting the strange behavior above in `gatsby develop`. When I had tried `gatsby build`, I got a different failure:

```
failed Building production JavaScript and CSS bundles - 38.768s

ERROR #98123  WEBPACK

Generating JavaScript bundles failed

Unexpected token (1:43)

File: src/vendor/foo/bar.js:1:43
```

(I'm not sure if 1:43 is supposed to be the line/column or what--it does not correspond to the `import.meta` reference and the first line doesn't even have 43 characters.)

But with the Emscripten setting above, the generated code does not include any `import.meta` references and works fine with gatsby-plugin-workerize-loader.

Sorry for the noise--hope this is useful if someone else runs across something similar.
